src/condor_utils/read_user_log_state.cpp

```diff
@@ -588,23 +575,29 @@ ReadUserLogState::SetScoreFactor( enum ScoreFactors which, int factor )
 ReadUserLog::FileStatus
 ReadUserLogState::CheckFileStatus( int fd, bool &is_empty )
 {
-	StatWrapper	sb;
+	struct stat sb;
+	int rc = 0;
+
+	if (fd < 0 && m_cur_path.empty()) {
+		dprintf(D_FULLDEBUG, "StatFile: no file to stat\n");
+		return ReadUserLog::LOG_STATUS_ERROR;
+	}
 
 	if ( fd >= 0 ) {
-		(void) sb.Stat( fd );
+		rc = fstat(fd, &sb);
 	}
 
-	if ( m_cur_path.length() && !sb.IsBufValid() ) {
-		(void) sb.Stat( m_cur_path.c_str() );
+	if ( m_cur_path.length() && rc != 0 ) {
+		rc = stat(m_cur_path.c_str(), &sb);
```

Comment:
stat if fstat failed?  why?

Reply:
Excessive flexibility. ReadUserLogState() can be constructed with a file path or without one. If constructed with a file path, then CheckFileStatus() can be called with a negative fd and the file path is stat()d. Otherwise, every call to CheckFileStatus() must be given a valid fd value. The only user, ReadUserLog, always passes in its m_fd data member. I don't know if there are any valid cases where m_fd is negative and the ReadUserLogState's internal file path should be used. I maintained the behavior of the existing code.

Reply:
I wasn't clear.   if fd < 0,  then rc is initialized 0, then fstat is skipped so stat is also skipped because of the rd != 0 test.  the only time stat is called is when fstat is called and fails.

Reply:
Ah, I see the problem now. I'll push a fix.